lib/native/http.ts

```diff
@@ -18,31 +18,69 @@ import { NativeResource } from "./native_resource";
 import { ResourceSafe } from '../common/resource_safety';
 import { ClientBootstrap, ClientTlsContext, SocketOptions } from './io';
 
-type ConnectionCallback = (error_code: Number) => void;
-
 export class HttpConnection extends NativeResource implements ResourceSafe {
 
-    constructor(
+    protected constructor(native_handle: any)
+    {
+        super(native_handle);
+    }
+
+    close() {
+        crt_native.http_connection_close(this.native_handle());
+    }
+}
+
+type ClientConnectionCallback = (connection: HttpClientConnection, error_code: Number) => void;
```

Comment:
Just a note: The bindings need something like an ErrorCode class so they can make any sense of these ints. I started writing one yesterday but put it on the backburner when I realized I should focus on HTTP first.

Reply:
I added an Error class that converts int -> string